rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/Costs.scala

```diff
@@ -88,5 +88,8 @@ trait Costs {
     def storageCost: Cost = Cost(a.map(a => a.serializedSize).sum)
   }
 
-  final val MAX_VALUE = PhloLimit(Long.MaxValue)
+  // Even though we use Long as phlo limit we pass in less than maximum value
+  // Because when we pass Long.MaxValue and we refund the deploy for clearing the tuplespace
+  // phlo limit overflows causing the deploy to throw OutOfPhlogistonError.
```

Comment:
Seems like a missing `toIntExact` at the place where it overflows. We should fix that.

Reply:
Also, this just begs for bugs. I'd either make phloLimit an Interger, or change casper code to use Longs.

Reply:
This doesn't solve the problem. If casper used  integers it would be the same. Not to mention that phlo limit has to be Long. In normal cases this won't ever be the problem because the deployer would need to have equivalent of `Long.MaxValue` REVs in the wallet and get refunded.

Reply:
I don't understand the problem then from how you described it in the comment.

When do we pass Long.MaxValue?
Why does a refund cause an overflow (I thought we only refund what has been spent, not more?)

BTW 'clearing the tuplespace' suggests clearing everything in the tuplespace. Is that what you tried to convey?

Reply:
I've reworded the comment. Is this better?

Reply:
Yes, the failure was in the test but tests delegate setting initial phlo limit to this value.

Reply:
> Either the refund is done by production code, in which case an overflow should mean a legitimate test failure or a bad setup

How will you differentiate between `Cost(-1)`, `Cost(-100)` or `Cost(Long.MinValue)` ? All are valid values but one is less probable than others.